packages/rule-schema-to-typescript-types/src/index.ts

```diff
@@ -69,9 +69,7 @@ function compileSchema(
 
   const refMap = new Map<string, string>();
   // we only support defs at the top level for simplicity
-  const defs = (schema.$defs ?? schema.definitions) as
-    | Record<string, JSONSchema4>
-    | undefined;
```

Comment:
IIRC I did this intentionally because both of the keys were not techincally `undefined` - is that not the case any more?

Reply:
Yup! `no-unnecessary-type-assertion` is complaining on them. Both `schema.$defs` and `schema.definitions` are `Record<string, JSONSchema4> | undefined`.